fetch/api/abort/general.js

```diff
@@ -0,0 +1,288 @@
+        signal._addAbortCallback(() => req.abort());
+      }
+    });
+  });
+}
+//*/
+
+function assert_abort_error(err) {
+  assert_equals(err.constructor, DOMException);
+  assert_equals(err.name, 'AbortError');
+}
+
+promise_test(async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  controller.abort();
+
+  await fetch('../resources/data.json', { signal }).then(() => {
+    assert_unreached("Fetch must not resolve");
+  }, err => {
+    // Using .catch rather than try/catch to ensure the promise
+    // is rejecting (rather than throwing).
+    assert_abort_error(err);
+  });
+}, "Aborting rejects with AbortError");
+
+promise_test(async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  controller.abort();
+
+  const log = [];
+
+  await Promise.all([
+    fetch('../resources/data.json', { signal }).then(
+      () => log.push('fetch-resolve'),
+      () => log.push('fetch-reject')
+    ),
+    Promise.resolve().then(() => log.push('next-microtask'))
+  ]);
+
+  assert_array_equals(log, ['fetch-reject', 'next-microtask']);
+}, "Already aborted signal rejects immediately");
+
+promise_test(async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  controller.abort();
+
+  const request = new Request('../resources/data.json', {
+    signal,
+    method: 'POST',
+    body: 'foo',
+    headers: { 'Content-Type': 'text/plain' }
+  });
+
+  await fetch(request).catch(() => {});
+
+  assert_false(request.bodyUsed, "Body has not been used");
+  assert_equals(await request.text(), 'foo', "Correct body");
+}, "Request is not 'used' if signal is aborted before fetching");
+
+const bodyMethods = ['arrayBuffer', 'blob', 'formData', 'json', 'text'];
+
+for (const bodyMethod of bodyMethods) {
+  promise_test(async () => {
+    const controller = new AbortController();
+    const signal = controller.signal;
+    
+    const log = [];
+    const response = await fetch('../resources/data.json', { signal });
+    
+    controller.abort();
+  
+    await Promise.all([
+      response[bodyMethod]().then(
+        () => log.push(`${bodyMethod}-resolve`),
+        err => {
+          assert_abort_error(err);
+          log.push(`${bodyMethod}-reject`);
+        }
+      ),
+      Promise.resolve().then(() => log.push('next-microtask'))
+    ]);
+  
+    assert_array_equals(log, [`${bodyMethod}-reject`, 'next-microtask']);
+  }, `response.${bodyMethod}() rejects if already aborted`);
+}
+
+promise_test(async () => {
+  const controller = new AbortController();
+  const signal = controller.signal;
+  const key = token();
+  controller.abort();
+
+  await fetch(`../resources/infinite-slow-response.py?key=${key}`, { signal }).catch(() => {});
+  
+  // I'm hoping this will give the browser enough time to (incorrectly) make the request
+  // above, if it intends to.
+  await fetch('../resources/data.json').then(r => r.json());
```

Comment:
Not really.  Doing another operation that takes the same path is what we do other places in wpt to approximate this.